network/p2p/gossip/gossip.go

```diff
@@ -20,27 +20,53 @@ import (
 	"github.com/ava-labs/avalanchego/utils/wrappers"
 )
 
+var (
+	_ Gossiper = (*ValidatorGossiper)(nil)
+	_ Gossiper = (*PullGossiper[testTx, *testTx])(nil)
+)
+
+// Gossiper gossips Gossipables to other nodes
+type Gossiper interface {
+	// Gossip runs a cycle of gossip. Returns an error if we failed to gossip.
+	Gossip(ctx context.Context) error
+}
+
 // GossipableAny exists to help create non-nil pointers to a concrete Gossipable
 // ref: https://stackoverflow.com/questions/69573113/how-can-i-instantiate-a-non-nil-pointer-of-type-argument-with-generic-go
 type GossipableAny[T any] interface {
 	*T
 	Gossipable
 }
 
+// ValidatorGossiper only calls [Gossip] if the given node is a validator
+type ValidatorGossiper struct {
+	Gossiper
+
+	NodeID     ids.NodeID
+	Validators p2p.ValidatorSet
+}
+
+func (v ValidatorGossiper) Gossip(ctx context.Context) error {
+	if !v.Validators.Has(ctx, v.NodeID) {
```

Comment:
This does not interfere with push gossip that may otherwise be going on, right?

Reply:
The bloom filter gossip stuff is sandboxed in its own implementation so it runs in parallel to the existing push gossip. The push gossip stuff also doesn't implement the `Gossiper` interface so it shouldn't ever be affected by this.